pkg/webhook/webhook.go

```diff
@@ -124,7 +124,7 @@ func getAdmissionResponseForBuild(ar *v1beta1.AdmissionReview, quayIntegration *
 
 	quayRegistryHostname, err := quayIntegration.GetRegistryHostname()
 
-	if build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
+	if buildv1.JenkinsPipelineBuildStrategyType == build.Spec.Strategy.Type || build.Spec.CommonSpec.Output.To == nil || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
```

Comment:
If I read the [docs](https://docs.openshift.com/container-platform/4.6/builds/managing-build-output.html) correct it seems like only the strategies (`build.spec.strategy.type`) of type `sourceStrategy` or `dockerStrategy` or `customStrategy` qualify for build outputs to be images. So maybe a reverse check on them is more future-proof than intercepting Jenkins builds. To be safe I'd also check if the `spec.output` element exists.

Reply:
Updated the PR